spacelift/data_azure_integrations_test.go

```diff
@@ -0,0 +1,42 @@
+package spacelift
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	. "github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/testhelpers"
+)
+
+func TestAzureIntegrationsData(t *testing.T) {
+	t.Run("when looking up integrations", func(t *testing.T) {
+		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
+
+		resourceName := "spacelift_azure_integration.test"
+		datasourceName := "data.spacelift_azure_integrations.test"
+
+		testSteps(t, []resource.TestStep{{
+			Config: fmt.Sprintf(`
+			resource "spacelift_azure_integration" "test" {
+				name                    = "test-integration-%s"
+				tenant_id               = "tenant-id"
+				default_subscription_id = "subscription-id"
+				labels                  = ["one", "two"]
+				space_id 				= "root"
+			}
+
+			data "spacelift_azure_integrations" "test" {
+				depends_on = [spacelift_azure_integration.test]
+			}
+			`, randomID),
+			Check: resource.ComposeTestCheckFunc(
+				Resource(datasourceName, Attribute("id", IsNotEmpty())),
+				CheckIfResourceNestedAttributeContainsResourceAttribute(datasourceName, []string{"integrations", "integration_id"}, resourceName, "integration_id"),
```

Comment:
Can we check if the values are correct, not just that they exist? Also, is there a reason you're not checking `default_subscription_id` and `labels`?